013-generalize-address-spaces.md

```diff
@@ -2,429 +2,372 @@
-> having access to independent storage. For these reasons they may have
-> properties that do not allow them to be viewed as part of the unified global
-> virtual address space accessible by all threads._
-> 
-> [non-normative] _It is target architecture specific whether multiple DWARF
-> address spaces are supported and how source language memory spaces map to
-> target architecture specific DWARF address spaces. A target architecture may
-> map multiple source language memory spaces to the same target architecture
-> specific DWARF address class. Optimization may determine that variable
-> lifetime and access pattern allows them to be allocated in faster scratchpad
-> memory represented by a different DWARF address space than the default for
-> the source language memory space._
-> 
-> Although DWARF address space identifiers are target architecture specific,
-> `DW_ASPACE_none` is a common address space supported by all target
-> architectures, and defined as the target architecture default address space.
-> 
-> DWARF address space identifiers are used by:
-> 
-> * The `DW_AT_address_space` attribute.
-> 
-> * The DWARF expressions: `DW_OP_aspace_bregx`,
->   `DW_OP_form_aspace_address`, `DW_OP_aspace_implicit_pointer`, and
->   `DW_OP_xderef*`.
-> 
-> * The CFI instructions: `DW_CFA_def_aspace_cfa` and `DW_CFA_def_aspace_cfa_sf`.
-
-[For further discussion]
-May want to clarify what `DW_AT_address_class` is for, now that the x86 examples
-have been removed, in a separate issue.
-
-In Section 3.7 "Memory Locations", add the following at the
-end of the first paragraph:
-
-> `DW_ASPACE_none` is defined as the target architecture default address space.
-> See 2.x Address Spaces.
+>    2.12 Address Spaces
+>
+>    DWARF address spaces correspond to target architecture specific
+>    linearly addressable memory areas. They are used in DWARF
+>    expression locations to describe in which target
+>    architecture specific memory area data resides.
+>
+>    *Target architecture specific DWARF address spaces may correspond
+>    to hardware supported facilities such as memory utilizing base
+>    address registers, scratchpad memory, and memory with special
+>    interleaving.  The size of addresses in these address spaces may
+>    vary. Their access and allocation may be hardware managed with
+>    each thread or group of threads having access to independent
+>    storage. For these reasons they may have properties that do not
+>    allow them to be viewed as part of the unified global virtual
+>    address space accessible by all threads.*
+>
+>    *It is target architecture specific whether multiple DWARF
+>    address spaces are supported and how source language memory
+>    spaces map to target architecture specific DWARF address
+>    spaces. A target architecture may map multiple source language
+>    memory spaces to the same target architecture specific DWARF
+>    address class. Optimization may determine that variable lifetime
```

Comment:
"address class" -> "address space"

The original proposal did define the separate concepts of address space (for the target architecture implementations) and address classes (for encoding of addresses).